x-pack/plugin/esql/qa/testFixtures/src/main/resources/inlinestats.csv-spec

```diff
@@ -653,9 +653,27 @@ required_capability: join_planning_v1
 | LIMIT 4
 ;
 
-emp_no:integer | salary:integer |  hire_date:datetime  | avg_salary:double
-         10001 |          57305 | 1986-01-01T00:00:00Z | 43869.63636363636
-         10002 |          56371 | 1985-01-01T00:00:00Z | 51831.818181818184
-         10003 |          61805 | 1986-01-01T00:00:00Z | 43869.63636363636
-         10005 |          63528 | 1989-01-01T00:00:00Z | 53487.07692307692
+emp_no:integer | salary:integer | avg_salary:double |  hire_date:datetime
+         10001 |          57305 | 43869.63636363636 | 1986-01-01T00:00:00Z
+         10002 |          56371 | 51831.818181818184| 1985-01-01T00:00:00Z
+         10003 |          61805 | 43869.63636363636 | 1986-01-01T00:00:00Z
+         10005 |          63528 | 53487.07692307692 | 1989-01-01T00:00:00Z
+;
+
+groupByRenamedColumn-Ignore
+required_capability: inlinestats_v4
+FROM employees
+| KEEP emp_no, languages, gender
+| INLINESTATS max_lang = MAX(languages) BY y = gender
```

Comment:
I discovered this issue while testing different variants of the order of columns.
In this particular case where `gender` is simply "aliased" with a different name, the query doesn't work, even though things more complex like `BY y = concat(gender, "")` do work. The investigation was more involved and I chose to tackle this one in a subsequent PR. Preliminary work show that `ReplaceAggregateNestedExpressionWithEval` and `CombineProjections` seem to work one against the other. The former adds an `eval` while the latter eliminates it and puts it back in the groupings of the aggregation.